feat: sync meta discussion

vinckr opened the pull request:
Added a check to sync.sh to distinguish between ory main projects, 
who have their own discussion board and 
/meta/discussions that is for all other projects. 

Had to change the variables to $project (lowercase), since the substitution was case sensitive. 

Changed the variables in the contributing template accordingly. 

Also closes https://github.com/ory/meta/pull/76.

aeneasr commented on scripts/sync.sh:
```diff
@@ -44,7 +44,11 @@ function sync {
     for f in $(find "$workdir/.github/ISSUE_TEMPLATE" "$workdir/.github/pull_request_template.md" "$workdir/CONTRIBUTING.md" "$workdir/SECURITY.md" "$workdir/CODE_OF_CONDUCT.md" "$workdir/LICENSE" -type f -print); do
-        env -i REPOSITORY="$project" PROJECT="$humanName" /bin/bash -c "envsubst < \"$f\" | sponge \"$f\""
+        if [[ "$project" = "hydra" || $project = "kratos" || $project = "oathkeeper" || $project = "keto" ]]; then
+        env -i discussions="$discussions" repository="$project" project="$humanName" /bin/bash -c "envsubst < \"$f\" | sponge \"$f\""
```
Please format and indent.

```suggestion
           env -i discussions="$discussions" repository="$project" project="$humanName" /bin/bash -c "envsubst < \"$f\" | sponge \"$f\""
```

aeneasr commented on scripts/sync.sh:
```diff
@@ -44,7 +44,11 @@ function sync {
+        if [[ "$project" = "hydra" || $project = "kratos" || $project = "oathkeeper" || $project = "keto" ]]; then
+        env -i discussions="$discussions" repository="$project" project="$humanName" /bin/bash -c "envsubst < \"$f\" | sponge \"$f\""
+        else
+        env -i discussions="https://github.com/meta/discussions" repository="$project" project="$humanName" /bin/bash -c "envsubst < \"$f\" | sponge \"$f\""
```
Please format and indent.

```suggestion
            env -i discussions="https://github.com/meta/discussions" repository="$project" project="$humanName" /bin/bash -c "envsubst < \"$f\" | sponge \"$f\""
```

aeneasr reviewed (changes requested):
This PR can not be accepted, it changes the environment variables from upper to lower case, but neglects to update all the references in the files updated by the `env` calls.

Please revert the lower case changes. Environment variables are upper case, not lower case.

vinckr commented:
Changed the env variables to uppercase, now I know the convention 👍 .

I changed all variables in CONTRIBUTING.md, and also config.yml

In 
".github/ISSUE_TEMPLATE" 
".github/pull_request_template.md" 
"SECURITY.md" 
"CODE_OF_CONDUCT.md" 
"LICENSE"
I could not find any variables that are being replaced. 

Am I looking in the wrong spot (templates/repository/common)?

aeneasr commented on scripts/sync.sh:
```diff
@@ -44,7 +44,11 @@ function sync {
     for f in $(find "$workdir/.github/ISSUE_TEMPLATE" "$workdir/.github/pull_request_template.md" "$workdir/CONTRIBUTING.md" "$workdir/SECURITY.md" "$workdir/CODE_OF_CONDUCT.md" "$workdir/LICENSE" -type f -print); do
-        env -i REPOSITORY="$project" PROJECT="$humanName" /bin/bash -c "envsubst < \"$f\" | sponge \"$f\""
+        if [[ "$project" = "ory/hydra" || $project = "ory/kratos" || $project = "ory/oathkeeper" || $project = "ory/keto" ]]; then
+          env -i Discussions="$discussions" Repository="$project" Project="$humanName" /bin/bash -c "envsubst < \"$f\" | sponge \"$f\""
```
As discussed, use UPPERCASE as it was before.

aeneasr commented:
Thank you, did you test the latest changes properly?

vinckr commented:
I tested them as you suggested with 
f=path/to/file 
and then the 
```
if .... 
else
fi 
```
I couldnt test the whole script, but the parts that I touched worked.

vinckr commented:
Missed one change that broke the git commands at the end of CONTRIBUTING.
All gucci now!
